### PR TITLE
[PLATFORM-1083] Security Slider

### PR DESCRIPTION
**Pull request by timoxley**
On top of #699 because otherwise my environment doesn't work.

security-slider

### CODE REVIEW

File: app/src/userpages/components/StreamPage/Show/SecurityView/Slider.jsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+})
+
+/**
+ * Finds positions of selector elements (e.g. radio inputs) in its container
+ * and renders a track + stops between them.
+ *
+ * Why reading positions from DOM? The center points of the options we're drawing graphics
+ * between is dependent on the width of the label text, which we can't know without rendering.
+ * Without this, the rendered points wouldn't line up with the text or the text wouldn't align with edges of container.
+ * Simple to just less CSS do its thing and piggypack on top of the layout it produces.
```
Comment (juhah) [inline app/src/userpages/components/StreamPage/Show/SecurityView/Slider.jsx]: less = let

File: app/src/userpages/components/StreamPage/Show/SecurityView/index.jsx [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+}
+
+type Props = OwnProps & StateProps & DispatchProps
+
+/*
+ * The security slider manipulates two boolean fields on the Stream object.
+ * The slider setting maps to these booleans as follows:
+ * Basic:
+ * requireSignedData: false
+ * requireEncryptedData: false
```
Comment (juhah) [inline app/src/userpages/components/StreamPage/Show/SecurityView/index.jsx]: Don't know if I have an older e&e instance running but `requireEncryptedData` is not saved or returned but the stream endpoint. If you set the selection to "Encrypted", it does not remember the choice after refreshing the page.

**Comment by hpihkala** [comment]
@timoxley #699 merged

**Comment by timoxley** [comment]
@juhah 

> Line dots should probably have cursor: pointer since they are clickable
Fixed.

> I don't have access to the design but should the icon in the stream list chance according to selection (signed -> blue icon, encrypted -> green icon)

image

Fixed.

**Comment by timoxley** [comment]
> Compared to previous version, icons are not faded anymore when not selected (I assume on purpose).

@juhah was a bug, fixed

**Pull request merged**